src/ImageSharp/Processing/Processors/Convolution/ConvolutionProcessorHelpers.cs

```diff
@@ -83,4 +88,99 @@ internal static float[] CreateGaussianSharpenKernel(int size, float weight)
+        // M = | m21, m22, m23 |, and we want to find: R = | r1, r2, r3 | and C = | c2 |.
+        //     | m31, m32, m33 |                                                  | c3 |
+        //
+        // We essentially get the following system of linear equations to solve:
+        //
+        //   / a11 = r1c1
+        //   | a12 = r2c1
+        //   | a13 = r3c1
+        //   | a21 = r1c2                   a11     a12     a13       a11     a12     a13
+        //   / a22 = r2c2, which gives us: ----- = ----- = ----- and ----- = ----- = -----.
+        //   \ a23 = r3c2                   a21     a22     a23       a31     a32     a33
+        //   | a31 = r1c3
+        //   | a32 = r2c3
+        //   \ a33 = r3c3
+        //
+        // As we said, there are infinite solutions to this problem (provided the input matrix is in
+        // fact linearly separable), but we can look at the equalities above to find a way to define
+        // one specific solution that is very easy to calculate (and that is equivalent to all others
+        // anyway). In particular, we can see that in order for it to be linearly separable, the matrix
+        // needs to have each row linearly dependent on each other. That is, its rank is just 1. This
+        // means that we can express the whole matrix as a function of one row vector (any of the rows
+        // in the matrix), and a column vector that represents the ratio of each element in a given column
+        // j with the corresponding j-th item in the reference row. This same procedure extends naturally
+        // to lineary separable 2D matrices of any size, too. So we end up with the following generalized
+        // solution for a matrix M of size NxN (or MxN, that works too) and the R and C vectors:
+        //
+        //     | m11, m12, m13, ..., m1N |                                       | m11/m11 |
+        //     | m21, m22, m23, ..., m2N |                                       | m21/m11 |
+        // M = | m31, m32, m33, ..., m3N |, R = | m11, m12, m13, ..., m1N |, C = | m31/m11 |.
+        //     | ...  ...  ...  ...  ... |                                       |   ...   |
+        //     | mN1, mN2, mN3, ..., mNN |                                       | mN1/m11 |
+        //
+        // So what this algorithm does is just the following:
+        //   1) It calculates the C[i] value for each i-th row.
+        //   2) It checks that every j-th item in the row respects C[i] = M[i, j] / M[0, j]. If this is
+        //      not true for any j-th item in any i-th row, then the matrix is not linearly separable.
+        //   3) It sets items in R and C to the values detailed above if the validation passed.
+        for (int y = 1; y < height; y++)
+        {
+            float ratio = matrix[y, 0] / matrix[0, 0];
+
+            for (int x = 1; x < width; x++)
+            {
+                if (Math.Abs(ratio - (matrix[y, x] / matrix[0, x])) > 0.0001f)
```

Comment:
Using hardcoded epsilon might be problematic in many cases. Personally I prefer to make it a parameter with a default value or at least to work with library-wide constants proven to work well for the given library.

What are ImageSharp use-cases for this feature?

Reply:
Yeah we should normalise constants. There might be one in Numerics already.

I plan to use this as an underlying optimisation when exposing a general convolution API.